internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,400 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package fuzzbinder
+
+// This file provides the scraper module that helps to get the fuzzing
+// statistics from the ClusterFuzz and OSS-Fuzz reports.
+// Two buckets from ClusterFuzz and OSS-Fuzz are used.
+//
+// The first is gs://oss-fuzz-coverage which a coverage bucket.
+// Its structure is expected to include the following files:
+//
+//   {projectName}/fuzzer_stats/{date}/{fuzz-target}.json
+//   {projectName}/reports/{date}/linux/summary.json
+//   {projectName}/srcmap/{date}.json
+//
+// The expected date format is "YYYYMMDD" like "20221202"
+// and the expected projectName is lowercase like "oak".
+// {projectName}/fuzzer_stats/{date}/{fuzz-target}.json contains
+// the coverage summary for a fuzz-target generated by Clang and
+// {projectName}/reports/{date}/linux/summary.json contains the coverage
+// summary for the whole project. The coverage  summary contains fine-grained
+// data related to coverage metrics.
+// {projectName}/srcmap/{date}.json contains the commit hash for which
+// the coverage reports were generated on a given date.
+//
+// The second is gs://{projectName}-logs.clusterfuzz-external.appspot.com
+// which is a bucket containing the fuzzers logs. Its structure is expected
+// to include the following files:
+//
+//   {fuzzEngine}_{projectName}_{fuzz-target}/{fuzzengine}_{sanitizer}_{projectName}/{date}/{time}.log
+//
+// The expected date format is "YYYY-MM-DD" like "2022-12-05".
+// An example of this file path is:
+//   libFuzzer_oak_apply_policy/libfuzzer_asan_oak/2022-12-05/12:43:47:680110.log
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// CoverageBucket is the OSS-Fuzz Google Cloud Storage bucket containing
+// the coverage reports.
+const CoverageBucket = "oss-fuzz-coverage"
+
+// CoverageSummary contains the coverage report summary.
+type CoverageSummary struct {
+	Data    []CoverageSummaryData `json:"data"`
+	Type    any                   `json:"type"`
```

Comment:
Perhaps add a comment explaining that we don't care about the details of these fields, and that is why we use `any`.  Or is that obvious?

BTW, why do we even need to declare these fields? Does the marshaling/unmarshaling fail if these fields are not declared here?

Reply:
I removed them because they are not needed.